ds3-autogen-c/src/main/resources/templates/TypeEnumConstant.tmplt

```diff
@@ -0,0 +1,7 @@
+<#include "CopyrightHeader.tmplt"/>
+
+typedef enum {
+<#list enumConstants as enum>
+    ${enum.getName()},
```

Comment:
This method of printing the enum constants will leave an extra unwanted comma after the last enum constant.  I suggest creating a CHelper function to properly delimit these items similar to getEnumValues in https://github.com/SpectraLogic/ds3_autogen/blob/master/ds3-autogen-java/src/main/java/com/spectralogic/ds3autogen/java/helpers/JavaHelper.java (line 331).